docs/tutorials/NEB/neb_tutorial.md

```diff
@@ -121,77 +120,77 @@ The NEB part of CASTEP should pick up on the imposed ionic constraint(s) and you
  
 in the `.castep` output file.
 
+*[Shouldn't this be referenced in the main documentation, as it looks like a specific feature of the NEB code?]*
 
 ### Do I need climbing NEB?
 
-The climbing NEB method modifies the force on the highest-energy image (usually close to the middle of the band) such that it tries to climb *up* the barrier. 
-This is very useful to get accurate barrier estimates, but care must be taken when more complex pathways (i.e. with multiple maxima) are found.
+The climbing NEB method modifies the force on the highest-energy image (usually close to the middle of the band) *[? What's a band?]* such that it tries to climb *up* the barrier. 
+This is very useful to get accurate barrier estimates, but care must be taken when more complex pathways (i.e. with multiple maxima) are found.  *[Don't get this!]*
 
 In our example the path is quite straightforward and it makes sense to use the climbing image method. We therefore set:
 
 `TSSEARCH_NEB_CLIMBING: TRUE`
 
-in the .param file.
+in the .param file. *[This is the default surely? Perhaps clearer in main documentation giving an example of when you *would* set False]* 
 
 
 ## Running the NEB
 
 ### Parallelisation
-Because the NEB calculation essentially runs N parallel CASTEP calculations (where N is the number of images), we can make very efficient use of the high-performance computers. The way to do this is via "task farming". The idea is to first parallelise the NEB calculation over the N images evenly and only then to employ other forms of parallelisation (e.g. over k-point and bands) within each image. 
-
-For example, if one CASTEP calculation fits on one node, and you have 7 images, then you could run on a total of 7 nodes and set the keywords:
+Because the NEB calculation essentially involves *N* independent CASTEP calculations (where *N* is the number of images), we can make very efficient use of high-performance computers. This is done via "task farming", running the calculations for the *N* images independently. For example, if one CASTEP calculation fits on one node, and you have 7 images, then you could run on a total of 7 nodes by setting the keyword:
 
 `NUM_FARMS: 7`
 
 in the .param file. 
 
-In general, setting `NUM_FARMS` to the number of NEB images will lead to good performance, provided the available computing resources can be split neatly by this number.
+In general, setting `NUM_FARMS` to the number of NEB images will lead to good performance, provided the available computing resources can be split neatly by this number.  *[Don't follow the last phrase. Surely all you need are 7 nodes?]*
```

Comment:
I guess you might run into performance issues if you cannot request as many nodes as you have images. For example if you have just one node available, then setting NUM_FARMS to 7 would split the workload 7 ways on that node, which might well be inefficient. (?)

Reply:
OK. I think this needs reframing. If you don't set NUM_FARMS, are the calculations does sequentially, without trying to parallelise? 

In this case, the question really comes down to whether your HPC set-up is sufficiently large for task farming to be useful. The no brainer is N nodes are available. If you had a small calculation and large nodes (e.g. 48 processors), then you might set NUM_FARMS to 4 say? 

But this discussion might be best in the more general documentation. I think task farming is used for J calculations? @jryates

Reply:
```suggestion
A good rule of thumb is to aim for `NUM_FARMS` = number of NEB images. That way each 'farm' is responsible for one NEB image. Since each farm will effectively run a series of single-point calculations on each NEB image they are responsible for, the compute resources of that farm should be appropriately sized for such single-point calculations. Too small and you could get an out-of-memory error, too large and you might end up over-parallelising, leading to poor performance. 

This also assumes that your available compute resources can be neatly split into `NUM_FARMS`. The easy case is when each farm can run on a single node; i.e. NUM_FARMS = number of NEB images = number of nodes. However, if you only have 2 nodes available, you cannot choose `NUM_FARMS = 7` since your compute resources cannot be distributed evenly across the farms. The total number of cores available must be divisible by `NUM_FARMS`. 

If you choose `NUM_FARMS` < number of images, the N images will be distributed amongst the farms as evenly as possible. If the number of images is not a multiple of `NUM_FARMS`, some farms will end up being responsible for more images than others, leading to poor performance.

If you choose `NUM_FARMS` > number of images, then some farms will likely be sitting idle for a lot of the time, resulting in poor performance.
```

Reply:
L154: "However, if you only have 2 nodes available, you cannot choose `NUM_FARMS = 7` since your compute resources cannot be distributed evenly across the farms."
What if your nodes have 14 processors? I would only use "cannot" or "must not" for really Bad Things, "cannot" = this will give an unrecoverable error. "must not" = this is really bad. You might get an invalid answer. "should not" = this is really not recommended. On your head be it. Something like "it wouldn't make sense to" for something that was a bit silly but not jeopardising your calculation.

L156: "poor performance" -> "reduced overall performance" I would use "poor performance" to mean "this is a really dumb thing to do since your calculation will run a factor of N times more slowly". If you have 3 farms and 8 images, it's not going to be that bad surely? 

"the N images will be distributed amongst the farms as evenly as possible". Presumably the images are run sequentially on each farm, rather than competing?

Is not setting NUM_FARMS equivalent to NUM_FARMS = 1?

L158: wouldn't the "excess" farms have nothing to do (rather than "being idle for a lot of time")?  

It is probably worth highlighting that if you set NUM_FARMS = N then you probably want to request N x as much processor resource, i.e. if you would run a single-point calculation on 1 compute node, then you would probably want N nodes?

Reply:
I've updated my suggested changes -- hopefully that clears things up?